packages/api/controllers/auth.js

```diff
@@ -0,0 +1,177 @@
+      q: `{${query}}`,
+    }, (err, res) => {
+      if (err) return console.log('The API returned an error: ', err);
+      const { messages } = res.data;
+      if (!messages) {
+        console.log('no messages');
+      } else if (messages.length) {
+        const messageIds = [];
+        messages.forEach((message) => {
+          let nomName;
+          const { id } = message;
+          messageIds.push(id);
+          const attachmentIds = [];
+          gmail.users.messages.get({
+            userId: 'me',
+            id: message.id,
+          }, (err, res) => {
+            if (err) return console.log('error retrieving message', err);
+            if (res.data.payload.headers) {
+              for (let i = 0; i < res.data.payload.headers.length; i++) {
+                if (res.data.payload.headers[i].name === 'Subject') {
+                  const confirmAwaiting = (goal, nomNames) => {
+                    let confirmResults;
+                    nomNames.forEach((app) => {
+                      if (goal.includes(app)) {
+                        confirmResults = app;
+                      }
+                    });
+                    return confirmResults;
+                  };
+                  const confirmedResults = confirmAwaiting(res.data.payload.headers[i].value, nomNames);
+                  if (confirmedResults) {
+                    nomName = confirmedResults;
+                  }
+                }
+              }
+            }
+            if (res.data.payload.parts.length >= 1) {
+              for (let i = 0; i < res.data.payload.parts.length; i++) {
+                if (res.data.payload.parts[i].body.attachmentId) {
+                  attachmentIds.push(res.data.payload.parts[i].body.attachmentId);
+                }
+              }
+            }
+            if (attachmentIds.length > 0) {
+              attachmentIds.forEach((attachmentId) => {
+                gmail.users.messages.attachments.get({
+                  userId: 'me',
+                  messageId: message.id,
+                  id: attachmentId,
+                }, (err, res) => {
+                  if (err) return console.log(err);
+                  const markAsRead = function (messageId) {
+                    gmail.users.messages.modify({
+                      userId: 'me',
+                      id: messageId,
+                      resource: {
+                        removeLabelIds: ['UNREAD'],
+                      },
+                    }, (err) => {
+                      if (err) {
+                        console.log(err, 'Failed to mark email as read!');
+                      } else {
+                        console.log('Email marked as Read!');
+                      }
+                    });
+                  };
+                  uploadFile(nomName, res.data.data);
+                  markAsRead(message.id);
+                });
+              });
+            }
+          });
+        });
+      }
+    });
+  }
```

Comment:
Can we break some of this up a bit? Maybe make the callbacks independent functions/break out the attachment handling?